### PR TITLE
Don't double wrap nullable records

### PR DESCRIPTION
**Pull request by parlough**
Fixes https://github.com/dart-lang/dartdoc/issues/3606

### CODE REVIEW

**Review by srawlins — APPROVED** [review]
Thanks!

**Comment by parlough** [comment]
Thanks for taking a look @srawlins :) - Feel free to merge this and my other PRs when you're happy. I don't have commit access to this repo.

**Review by srawlins — APPROVED** [review]

**Pull request merged**
